### PR TITLE
add the Front

### PR DESCRIPTION
**Pull request by gOOvER**
# Description

The egg adds support for **The Front** to Pterodactyl. Feel free to buy me some Coffey, since it cost me a hell of a lot of nerves :D

fixes: #2504 

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/steamcmd_servers/thefront/egg-the-front.json [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
...all game using steamcmd. please wait...${NC}\"\r\necho -e \"${BLUE}-------------------------------------------------${NC}\"\r\n\r\n#rm -fR \/mnt\/server\/steamcmd\/linux32\/steamapps\/content\/app_2334200\/*\r\n\r\n.\/steamcmd.sh +force_install_dir \/mnt\/server +login anonymous +app_update 1007 validate +quit\r\n.\/steamcmd.sh +force_install_dir \/mnt\/server +login anonymous +download_depot 2334200 2334201 +quit\r\necho -e \"${GREEN}..done..${NC}\"\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\n## add below your custom commands if needed\r\n\r\nrm -fR \/mnt\/server\/ProjectWar\r\nrm -fR \/mnt\/server\/Engine\r\nmv -f \/mnt\/server\/steamcmd\/linux32\/steamapps\/content\/app_2334200\/depot_2334201\/* \/mnt\/server\r\n\r\n## install end\r\necho -e \"${BLUE}-------------------------------------------------${NC}\"\r\necho -e \"${GREEN}Installation completed...${NC}\"\r\necho -e \"${BLUE}-------------------------------------------------${NC}\"",
```
Comment (dagbs) [inline game_eggs/steamcmd_servers/thefront/egg-the-front.json:27]: `rm -fR /mnt/server/ProjectWar`
should probably backup the `/mnt/server/ProjectWar/saved` folder before deleting it

Comment (gOOvER) [reply game_eggs/steamcmd_servers/thefront/egg-the-front.json:27]: No, because i use an own dir for the config ;)

**Comment by gOOvER** [comment]
Everyone screams for the game, but no one tests it; the main thing is to have the egg :/

**Comment by ValhallaCommunityOfficial** [comment]
Thank you for the egg! However, I'm getting "The [SERVER] PvE variable field must be true or false." No matter what I put in the box.
I want to turn PVE on so ive tried "true" "on" "PvE"

**Comment by gOOvER** [comment]
> Thank you for the egg! However, I'm getting "The [SERVER] PvE variable field must be true or false." No matter what I put in the box. I want to turn PVE on so ive tried "true" "on" "PvE"
> 
> Edit: Got it to work. Value must be "1" or "0"

Field is Boolean, so only 1 or 0 works.

**Comment by ValhallaCommunityOfficial** [comment]
The server shows in the server list but when I try to join it never fully connects.
[ProjectWar.log](https://github.com/parkervcp/eggs/files/12909507/ProjectWar.log)

**Comment by gOOvER** [comment]
> The server shows in the server list but when I try to join it never fully connects. [ProjectWar.log](https://github.com/parkervcp/eggs/files/12909507/ProjectWar.log)

What means it never fully connects?

**Comment by ValhallaCommunityOfficial** [comment]
> > The server shows in the server list but when I try to join it never fully connects. [ProjectWar.log](https://github.com/parkervcp/eggs/files/12909507/ProjectWar.log)
> 
> What means it never fully connects? did you allocate more than 8GB RAM?

Apologies for not being more descriptive.
I have only allocated 12gb to test, it's only using 6gb ram when launched.
When I click the server in the server list and click "connect" there's no option to create character and 30 seconds later it says "Server Connection has expired!"

**Comment by gOOvER** [comment]
> > > The server shows in the server list but when I try to join it never fully connects. [ProjectWar.log](https://github.com/parkervcp/eggs/files/12909507/ProjectWar.log)
> > 
> > 
> > What means it never fully connects? did you allocate more than 8GB RAM?
> 
> Apologies for not being more descriptive. I have only allocated 12gb to test, it's only using 6gb ram when launched. When I click the server in the server list and click "connect" there's no option to create character and 30 seconds later it says "Server Connection has expired!"

That sounds more like not all ports open

**Comment by ValhallaCommunityOfficial** [comment]
> > > > The server shows in the server list but when I try to join it never fully connects. [ProjectWar.log](https://github.com/parkervcp/eggs/files/12909507/ProjectWar.log)
> > > 
> > > 
> > > What means it never fully connects? did you allocate more than 8GB RAM?
> > 
> > 
> > Apologies for not being more descriptive. I have only allocated 12gb to test, it's only using 6gb ram when launched. When I click the server in the server list and click "connect" there's no option to create character and 30 seconds later it says "Server Connection has expired!"
> 
> That sounds more like not all ports open. Their discord is full of this error message and in 99% its a firewall problem.
> 
> My server runs now over 24h with a lot of restarts without any problem :)

Ah, thank you! that fixed it. I had setup the ports wrong. My noob self was using the game port as the beacon port lol
All is good now, everything works!

**Comment by dagbs** [comment]
> Everyone screams for the game, but no one tests it; the main thing is to have the egg :/

Worked for me 100%

I had an issue with my egg (zombies wouldn't move) so your egg fixed all of that. :D

**Comment by gOOvER** [comment]
> Memory: 9.14GB
How many players?

**Comment by dagbs** [comment]
> > Memory: 9.14GB
> 
> How many players?

1, me (testing)

**Comment by scarecr0w12** [comment]
Getting the connection expired issue too with this egg, can install manually on server and it runs just fine on same machine and same ports/ip, but Egg does not. Though likly that its a server files issue given the problems thus far with the updates.

**Comment by gOOvER** [comment]
> Getting the connection expired issue too with this egg, can install manually on server and it runs just fine on same machine and same ports/ip, but Egg does not. Though likly that its a server files issue given the problems thus far with the updates.

see hat i wrote above about this error; its 99,99999999% problem with your ports and firewall

(and last time now, taht i answer to this error)

**Comment by scarecr0w12** [comment]
I can tell you right now, its not a ports or firewall issue, Firewall is completely off.

**Comment by scarecr0w12** [comment]
And as mentioned i can run this server fine without using pterodactyl on same hardware, same IP and same port

**Comment by gOOvER** [comment]
> I can tell you right now, its not a ports or firewall issue, Firewall is completely off.

Then check your ports or when you host at home, your network; its definityly no egg issue

the egg runs since yesterday for 2 Hosters, now on over 60 Client VMs without any problem.

**Comment by gOOvER** [comment]
> And as mentioned i can run this server fine without using pterodactyl on same hardware, same IP and same port

then I would recommend you not to use the egg and run it on your VM. This is not the right place to debug your ptero installation.

By the way you didn't give any infos, like logs ;)

**Comment by scarecr0w12** [comment]
Server only runs by adding ?udrs=steam at the end of `ProjectWar_Start?DedicatedServer?MaxPlayers={{MAX_PLAYERS}}?udrs=steam` 

My ptero install is perfectly fine

**Comment by gOOvER** [comment]
> Server only runs by adding ?udrs=steam at the end of `ProjectWar_Start?DedicatedServer?MaxPlayers={{MAX_PLAYERS}}?udrs=steam`
> 
> My ptero install is perfectly fine

This is an old flag and should not be used since last update. So you host at home?

**Comment by scarecr0w12** [comment]
No, i have a dedicated server, The issue is that the server is only connectable when that is added, other wise it goes connection expired, as mentioned, only when using this egg

**Comment by dagbs** [comment]
image
image

fresh install took ~20min
only changed the ports as needed to what I had available
server name `pterodactyl fresh gOOvER egg by dagbs`

so yes, it works...

**Comment by gOOvER** [comment]
> No, i have a dedicated server, The issue is that the server is only connectable when that is added, other wise it goes connection expired, as mentioned, only when using this egg

as i said; thats not the place to debug your server. Egg worlks fine. Add tthe flag if you want. I will not add this flag, because its not needed, only when you host at home

**Comment by scarecr0w12** [comment]
> image image
> 
> fresh install took ~20min only changed the ports as needed to what I had available server name `pterodactyl fresh gOOvER egg by dagbs`
> 
> so yes, it works...

0ms? looks like you're hosting at home.

**Comment by ValhallaCommunityOfficial** [comment]
For those having trouble with ports here's an example of my server ports setup.

Game Port 25567
[SERVER] BEACONPORT 25568
[SERVER] QUERYPORT 25577
[SERVER] SHUTDOWNSERVICEPORT 25577

Which I'm not 100% sure if this is correct but I am able to join my own dedicated server that is hosted in another state. The only issue I'm having is when I'm playing, battlemetrics still shows 0 players playing.

If someone can confirm if this looks correct that would be great, thank you.

**Comment by gOOvER** [comment]
Battlemetrics is not part of the egg and i think, its more a game problem

**Comment by GameAreaCZ** [comment]
Just really fast tried this egg. Fast installation, not needed any changes ony set the variables click start. Server is connectable just fine. Will be switching from the Windows version to this in few days to test it more. Thank you for it.

**Comment by dagbs** [comment]
> EDIT // Only thing I changed is the docker image to the Quinten's one, because that one shutting down the server safely.

Safely? There is no RCON (at least not yet) to pass in commands; so, how are they doing it?

**Comment by QuintenQVD0** [comment]
> Just really fast tried this egg. Fast installation, not needed any changes ony set the variables click start. Server is connectable just fine. Will be switching from the Windows version to this in few days to test it more. Thank you for it.
> 
> EDIT // Only thing I changed is the docker image to the Quinten's one, because that one shutting down the server safely.

Non offical image are not allowed in offical eggs.

**Comment by QuintenQVD0** [comment]
> > EDIT // Only thing I changed is the docker image to the Quinten's one, because that one shutting down the server safely.
> 
> Safely? There is no RCON (at least not yet) to pass in commands; so, how are they doing it?

By default because rhe wings logic is a litte broken SIGKILL is send what is force exit. This images makes ^C send SIGINT what is ctrl c, so force exit but grafcefully.

**Comment by gOOvER** [comment]
Lastest cThe last commit changes the way the server is updated. 

instead of downloading the depot, the server is installed and updated again via the normal way.

Autoupdate can be used againmmit change the way, how the server is

**Comment by dagbs** [comment]
Windows is default with the new installer?

**Review by QuintenQVD0 — APPROVED** [review]
LGTM! thanks

**Pull request merged**